src/main/docker/patches/issue-548-v5.0.5.patch

```diff
@@ -0,0 +1,53 @@
+diff --git a/src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java b/src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java
+index c4e2d413a..2c7d54ddb 100644
+--- a/src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java
++++ b/src/main/java/edu/illinois/library/cantaloupe/image/Metadata.java
+@@ -5,7 +5,9 @@
+ import com.fasterxml.jackson.annotation.JsonInclude;
+ import com.fasterxml.jackson.annotation.JsonProperty;
+ import edu.illinois.library.cantaloupe.Application;
++import edu.illinois.library.cantaloupe.image.exif.DataType;
+ import edu.illinois.library.cantaloupe.image.exif.Directory;
++import edu.illinois.library.cantaloupe.image.exif.Field;
+ import edu.illinois.library.cantaloupe.image.exif.Tag;
+ import edu.illinois.library.cantaloupe.image.iptc.DataSet;
+ import edu.illinois.library.cantaloupe.util.StringUtils;
+@@ -152,9 +154,17 @@ public Orientation getOrientation() {
+     }
+ 
+     private void readOrientationFromEXIF() {
++        Field field = exif.getField(Tag.ORIENTATION);
+         Object value = exif.getValue(Tag.ORIENTATION);
+-        if (value != null) {
+-            orientation = Orientation.forEXIFOrientation((int) value);
++        if (field != null && value != null) {
++            switch (field.getDataType()) {
++              case LONG:
++                orientation = Orientation.forEXIFOrientation(Math.toIntExact((long) value));
++                break;
++              case SHORT:
```

Comment:
I'm wondering if `case SHORT` should be the `default` (like the original code). I don't know (yet) if there are other significant values returned by `getDataType()` (or values that would be problematic and should be ignored). Will check on that tomorrow.

Reply:
Sounds smart.